main/include/deck.hpp

```
@@ -1,56 +1,137 @@
+            std::random_device rd;
+            std::mt19937 g(rd());
+            std::shuffle(state.cards.begin(), state.cards.end(), g);
+            // go back to idle
+            state.state = DeckState::IDLE;
+        }
     }
 
     // external transition
-    void externalTransition(deck_modelState& state, double e) const override {
+    void externalTransition(deckState& state, double e) const override {
         //your external transition function hoes here
+        std::vector<int> drawInPortMsg = drawInPort->getBag();
+        std::vector<int> shuffleInPortMsg = shuffleInPort->getBag();
+        if (!drawInPortMsg.empty()){
+            if (!state.cards.empty()){
+                state.state = DeckState::DRAW;
+            }
+        }
+        //if shuffle, shuffle deck and go idle
+        else if (!shuffleInPortMsg.empty()) {
+            if (!state.cards.empty()){
+                state.state = DeckState::SHUFFLE;
+            }
+        }
     }
     
     
     // output function
-    void output(const deck_modelState& state) const override {
+    void output(const deckState& state) const override {
         //your output function goes here
+        if (state.state == DeckState::DRAW && !state.cards.empty()){
+            int drawnCard = state.cards.back();
+            cardOutPort->addMessage({state.current_player, drawnCard});
+        }
+        else if (state.state == DeckState::SHUFFLE) {
+            hitOutPort->addMessage(1);
+        }
     }
 
     // time_advance function
-    [[nodiscard]] double timeAdvance(const deck_modelState& state) const override {     
-            return state.sigma;
+    [[nodiscard]] double timeAdvance(const deckState& state) const override {     
+        switch (state.state) {
+            case DeckState::DRAW: return 1;
+            case DeckState::SHUFFLE: return 10;
+            case DeckState::IDLE: return std::numeric_limits<double>::infinity();
+        }
+        return 1;
```

maybe we should raise an error? Could hide erroneous states if we get to here